log-warper/src/Log/Event.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Contains implementations of logging records and events which are consumed by 'LogAction'.
+
+module Log.Event
+       ( -- * Types
+         Event (..)
+       , eName
+       , eSeverity
+       , eMessage
+       , eConfiguration
+
+       , RichEvent (..)
+       , reEvent
+       , reThread
+       , reTime
+
+         -- * Functions
+       , makeRich
+       ) where
+
+import Universum
+
+import Control.Concurrent (ThreadId, myThreadId)
+import Data.Time.Clock (UTCTime, getCurrentTime)
+import Lens.Micro.Platform (makeLenses)
+
+import Log.Configuration (Configuration, Extension, cGlobal, gpShowTid, gpShowTime)
+import Log.Core.Action (LogAction (..), cbind)
+import Log.Name (LoggerName (..))
+import Log.Severity (Severity (..))
+
+
+{- | Bundle of all basic logging data. This logging record also contains
+configuration for logging. It can be used to process messages. All 'LogAction's
+in @log-warper@ should have form:
+
+@
+'LogAction' someM ('Event' exts)
+@
+-}
+data Event (exts :: [Extension]) = Event
+    { _eName          :: !LoggerName
+    , _eSeverity      :: !Severity
+    , _eMessage       :: !Text
```

Comment:
Do we want `Text` here? I suggest to think about `Data.Text.Lazy.Builder` here, like `katip` use (https://github.com/Soostone/katip/blob/master/katip/src/Katip/Core.hs#L184). possible it allow make logging more flexible and not depending on Text type here (be polymorphic against Text/String/Bytestring/...)

Reply:
@avnik That's definitely a thing worth considering. But I 'm not sure about this without benchmarks. It's better to create another issue, investigate this problem, implment benchmarks and rewrite in future it's needed. With structured logging support in future this will be not `Text` but some complex data type.

Reply:
That's definitely a thing _not_ worth considering for now. Same as the `Vector` question later. Unless we have the real problem with logger performance I suggest to abandon these considerations and concentrate on the API and general design.